src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/ArtifactUpdateApiClientImpl.java

```diff
@@ -119,7 +119,12 @@ private Client getClient(String login, String password) {
         ClientConfig config = new DefaultClientConfig();
         Client client = Client.create(config);
         client.setExecutorService(asyncRequestsExecutorService);
-        client.addFilter(new HTTPBasicAuthFilter(login, password));
+        if (login != null && !login.isEmpty() && password != null) {
```

Comment:
Since you already use guava, you can use: [Strings.isNullOrEmpty()](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Strings.html#isNullOrEmpty(java.lang.String))